src/overlays/actors/ovl_En_Fu_Kago/z_en_fu_kago.c

```diff
@@ -28,46 +39,371 @@ const ActorInit En_Fu_Kago_InitVars = {
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp34);
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+
+    while (npc != NULL) {
+        if (npc->id == ACTOR_EN_FU) {
+            this->dyna.actor.parent = npc;
+            break;
+        }
+        npc = npc->next;
+    }
+
+    if (npc == NULL) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+
+    Collider_InitSphere(globalCtx, &this->collider);
+    Collider_SetSphere(globalCtx, &this->collider, &this->dyna.actor, &sSphereInit);
+    this->collider.dim.worldSphere.radius = 10;
+    this->unk_33A = 0;
+    this->unk_33C = 0;
+    this->dyna.actor.colChkInfo.health = 1;
+    func_80ACF994(this);
+}
+
+void EnFuKago_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFuKago* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+s32 func_80ACF8B8(EnFuKago* this, GlobalContext* globalCtx) {
+    this->collider.dim.worldSphere.center.x = this->dyna.actor.world.pos.x;
+    this->collider.dim.worldSphere.center.y = this->dyna.actor.world.pos.y + 10.0f;
+    this->collider.dim.worldSphere.center.z = this->dyna.actor.world.pos.z;
+
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        this->collider.base.ocFlags1 &= ~OC1_HIT;
+
+        if (this->collider.base.oc->id == ACTOR_EN_BOM) {
+            this->dyna.actor.child = this->collider.base.oc;
+        } else if (this->collider.base.oc->id == ACTOR_PLAYER) {
+            func_80ACFA78(this, globalCtx);
+            return false;
+        } else {
+            return false;
+        }
+
+        Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_SY_TRE_BOX_APPEAR);
+        return true;
+    } else {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+    return false;
+}
+
+void func_80ACF994(EnFuKago* this) {
+    this->actionFunc = func_80ACF9A8;
+}
+
+void func_80ACF9A8(EnFuKago* this, GlobalContext* globalCtx) {
+    if (func_80ACF8B8(this, globalCtx)) {
+        func_80ACF9DC(this);
+    }
+}
+
+void func_80ACF9DC(EnFuKago* this) {
+    EnBom* bom = (EnBom*)this->dyna.actor.child;
+
+    bom->timer = 20;
+    this->actionFunc = func_80ACF9FC;
+}
+
+void func_80ACF9FC(EnFuKago* this, GlobalContext* globalCtx) {
+    EnBom* bom = (EnBom*)this->dyna.actor.child;
+    EnFu* fu = (EnFu*)this->dyna.actor.parent;
+
+    if (this->dyna.actor.child->update == NULL) {
+        this->dyna.actor.child = NULL;
+        func_80ACF994(this);
+    } else if (bom->timer == 1) {
+        fu->unk_548++;
+        func_80ACFA78(this, globalCtx);
+    } else {
+        this->dyna.actor.child->world.pos.x = this->dyna.actor.world.pos.x;
+        this->dyna.actor.child->world.pos.z = this->dyna.actor.world.pos.z;
+    }
+}
+
+void func_80ACFA78(EnFuKago* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f sp98;
+    Vec3s sp90;
+    Vec3s sp88;
```

Comment:
```suggestion
    Vec3s sp88 = D_80AD070C;
```
Also this looks like `D_80AD070C` was just declared on the stack?